### PR TITLE
Contact enhance

### PR DESCRIPTION
**Pull request by shushengfuqin**
# Description

add vender contact  domain, repository, handler, router.
in the repository there are query modification for findAll, findById, and update. they are made for vender. where in the query they will hard code the contact_type to vender. 
ex. findAll -> findAllVender 

Test are still in working.

### CODE REVIEW

**Review by instamine — APPROVED** [review]
VenderContactRouter and venderContactHandler looks good.

**Review by obonobo — CHANGES_REQUESTED** [review]

File: src/main/java/ca/serum390/godzilla/data/repositories/VenderContactRepository.java [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+    //Find all customer
+    @Modifying
+    @Query("SELECT * FROM contact WHERE contact_type = 'vender'")
+    Mono<Integer> findAllVender();
+
+    //Find by id where contact_type = customer
+    @Modifying
+    @Query("SELECT * FROM contact WHERE ID = $1 AND contact_type = 'vender'")
+    Mono<Integer> findByIdVender(int id);    
+    //Update sales contact where contact_type = vendor
```
[the remark this answers is not in the document]
Comment (shushengfuqin) [reply src/main/java/ca/serum390/godzilla/data/repositories/VenderContactRepository.java:24]: done

**Review by obonobo — APPROVED** [review]
Looks good, just clean up the comments

File: src/main/java/ca/serum390/godzilla/data/repositories/VenderContactRepository.java [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+    @Modifying
+    @Query("UPDATE contact SET COMPANY_NAME = $1, CONTACT_NAME = $2, ADDRESS= $3, CONTACT=$4, CONTACT_TYPE='vender'  WHERE ID = $5")
+    Mono<Integer> updateVender(
+            String companyName,
+            String contactName,
+            String address,
+            String contact,
+            String string, Integer id);
```
[the remark this answers is not in the document]
Comment (shushengfuqin) [reply src/main/java/ca/serum390/godzilla/data/repositories/VenderContactRepository.java:32]: it should be contactType. 
I took it out

File: src/main/java/ca/serum390/godzilla/data/repositories/VenderContactRepository.java [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+
+import org.springframework.data.r2dbc.repository.Modifying;
+import org.springframework.data.r2dbc.repository.Query;
+import org.springframework.data.repository.reactive.ReactiveCrudRepository;
+
+import ca.serum390.godzilla.domain.vender.VenderContact;
+import reactor.core.publisher.Mono;
+
+public interface VenderContactRepository extends ReactiveCrudRepository<VenderContact, Integer>{
+    /**
```
[the remark this answers is not in the document]
Comment (shushengfuqin) [reply src/main/java/ca/serum390/godzilla/data/repositories/VenderContactRepository.java:11]: @Repository added

File: src/main/java/ca/serum390/godzilla/data/repositories/VenderContactRepository.java [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+     * Query for vendor contact
+     */
+
+    //Find all customer
```
[the remark this answers is not in the document]
Comment (shushengfuqin) [reply src/main/java/ca/serum390/godzilla/data/repositories/VenderContactRepository.java:15]: done

File: src/main/java/ca/serum390/godzilla/api/handlers/VenderContactHandler.java [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+
+    public VenderContactHandler(VenderContactRepository venderContact) {
+        this.venderContacts = venderContact;
+    }
+    
+    /**
+     * API for vender
+     */
+
+    //get all vender contact
```
[the remark this answers is not in the document]
Comment (shushengfuqin) [reply src/main/java/ca/serum390/godzilla/api/handlers/VenderContactHandler.java:28]: done

File: src/main/java/ca/serum390/godzilla/api/handlers/VenderContactHandler.java [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+import org.springframework.stereotype.Component;
+import org.springframework.web.reactive.function.server.ServerRequest;
+import org.springframework.web.reactive.function.server.ServerResponse;
+
+import ca.serum390.godzilla.data.repositories.VenderContactRepository;
+import ca.serum390.godzilla.domain.vender.VenderContact;
+import reactor.core.publisher.Mono;
+
+@Component
+public class VenderContactHandler {
```
[the remark this answers is not in the document]
Comment (shushengfuqin) [reply src/main/java/ca/serum390/godzilla/api/handlers/VenderContactHandler.java:16]: done

File: src/main/java/ca/serum390/godzilla/api/handlers/VenderContactHandler.java (hunk 5 above)
[the remark this answers is not in the document]
Comment (shushengfuqin) [reply src/main/java/ca/serum390/godzilla/api/handlers/VenderContactHandler.java]: done

File: src/main/java/ca/serum390/godzilla/data/repositories/VenderContactRepository.java (hunk 4 above)
[the remark this answers is not in the document]
Comment (shushengfuqin) [reply src/main/java/ca/serum390/godzilla/data/repositories/VenderContactRepository.java]: done

**Pull request merged**